data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/compression/CompressionOption.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.s3.compression;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public enum CompressionOption {
```

Comment:
Is there a reason not to reuse and extend the one in data-prepper-plugins:common?

Reply:
Yes, the existing one has `automatic` which doesn't make sense for compression. And it supports `snappy` which is not yet available.

I'd like to find a way to better merge these in the future.